core/src/main/java/org/apache/iceberg/SerializedTable.java

```diff
@@ -0,0 +1,290 @@
+    this.defaultSpecId = defaultSpecId;
+    this.specs = specs;
+    this.defaultSortOrderId = defaultSortOrderId;
+    this.sortOrders = sortOrders;
+    this.io = io;
+    this.encryption = encryption;
+    this.locationProvider = locationProvider;
+  }
+
+  private Table lazyTable() {
+    if (lazyTable == null) {
+      synchronized (this) {
+        if (lazyTable == null) {
+          TableOperations ops = new StaticTableOperations(metadataFileLocation, io, locationProvider);
+          this.lazyTable = newTable(ops, name);
+        }
+      }
+    }
+
+    return lazyTable;
+  }
+
+  protected Table newTable(TableOperations ops, String tableName) {
+    return new BaseTable(ops, tableName);
+  }
+
+  private String metadataFileLocation(Table table) {
+    if (table instanceof HasTableOperations) {
+      TableOperations ops = ((HasTableOperations) table).operations();
+      return ops.current().metadataFileLocation();
+
+    } else if (table instanceof BaseMetadataTable) {
+      return ((BaseMetadataTable) table).metadataLocation();
+
+    } else {
+      throw new IllegalArgumentException("Cannot determine metadata file location for table " + table.name());
+    }
+  }
+
+  private <K, V> Map<K, V> copy(Map<K, V> map) {
+    Map<K, V> copy = Maps.newHashMap(map);
+    return Collections.unmodifiableMap(copy);
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  public String location() {
+    return location;
+  }
+
+  @Override
+  public Map<String, String> properties() {
+    return properties;
+  }
+
+  @Override
+  public Schema schema() {
+    return schema;
+  }
+
+  @Override
+  public PartitionSpec spec() {
+    return specs.get(defaultSpecId);
+  }
+
+  @Override
+  public Map<Integer, PartitionSpec> specs() {
+    return specs;
+  }
+
+  @Override
+  public SortOrder sortOrder() {
+    return sortOrders.get(defaultSortOrderId);
+  }
+
+  @Override
+  public Map<Integer, SortOrder> sortOrders() {
+    return sortOrders;
+  }
+
+  @Override
+  public FileIO io() {
+    return io;
+  }
+
+  @Override
+  public EncryptionManager encryption() {
+    return encryption;
+  }
+
+  @Override
+  public LocationProvider locationProvider() {
+    return locationProvider;
+  }
+
+  @Override
+  public void refresh() {
+    throw new UnsupportedOperationException(errorMsg("refresh"));
+  }
+
+  @Override
+  public TableScan newScan() {
+    return lazyTable().newScan();
+  }
+
+  @Override
+  public Snapshot currentSnapshot() {
+    throw new UnsupportedOperationException(errorMsg("currentSnapshot"));
+  }
+
+  @Override
+  public Snapshot snapshot(long snapshotId) {
+    throw new UnsupportedOperationException(errorMsg("snapshot"));
+  }
+
+  @Override
+  public Iterable<Snapshot> snapshots() {
+    throw new UnsupportedOperationException(errorMsg("snapshots"));
+  }
+
+  @Override
+  public List<HistoryEntry> history() {
+    throw new UnsupportedOperationException(errorMsg("history"));
+  }
```

Comment:
Not sure about this, but in the future it might be useful to implement these to be able to access the previous history/snapshots of the static Table.

Use-case: We have 2 Iceberg tables and would like to have the valid snapshot combinations for the past few days. If we need to serialize both tables and then do the job on a different worker node then we need to read the history for the serialized tables.

Any particular reason not to implement these as delegating them to `lazyTable()`?

Thanks,
Peter

Reply:
Went ahead and implemented it now. Good catch, @pvary. Resolving the thread.